app/controllers/async_info_controller.rb

```diff
@@ -42,7 +42,8 @@ def broadcast_data
   #       decorated version of the user.  It would be nice if we were using the same "variable" for
   #       the cache key and for that which we cache.
   def user_data
-    Rails.cache.fetch(user_cache_key, expires_in: NUMBER_OF_MINUTES_FOR_CACHE_EXPIRY.minutes) do
+    Rails.cache.fetch("#{current_user.cache_key_with_version}/user-info",
```

Comment:
This is a cleaner approach. 

It will produce cache key similar to `"users/13-20230208011546691068/user-info"`. The "version" that it includes is the updated_at, which will change in response to changes in last_sign_in, last_reacted_at, last_followed_at, etc. Even if this isn't the perfect answer, with the short expiration of 15 minutes will remedy it.

Reply:
What's the `last_sign_in, last_reacted_at, last_followed_at, etc` here? Are there columns we care about in the old cache key that wouldn't be represented here?

I definitely think of the existing system as basically a total shotgun approach to begin with, so I'm in favor of moving to a simpler, more reliable approach, but just want to flesh out this.

> I'm also pretty convinced that cache expiration may not even be necessary, but don't want to risk it.

Can you expand on this?

Reply:
Yeah, I was referring to the removed `def user_cache_key` which based the cache key on numerous attributes is covered by `#cache_key_with_version`. It's because many, if not all of those attributes, trigger touch callback on update, which in-turn touches `updated_at`. That is the one attribute `cache_key_with_version` uses as the version.

My hypothesis is explicit cache expiration may not be necessary because this cache will be invalidated frequently from common user actions. But I just realized that explicit cache expiration is a good idea to have as redundancy.